src/Workspaces/Core/Portable/RegularExpressions/RegexCharClass.cs

```diff
@@ -0,0 +1,350 @@
+            "IsCurrencySymbols",
+            "IsCyrillic",
+            "IsCyrillicSupplement",
+            "IsDevanagari",
+            "IsDingbats",
+            "IsEnclosedAlphanumerics",
+            "IsEnclosedCJKLettersandMonths",
+            "IsEthiopic",
+            "IsGeneralPunctuation",
+            "IsGeometricShapes",
+            "IsGeorgian",
+            "IsGreek",
+            "IsGreekExtended",
+            "IsGreekandCoptic",
+            "IsGujarati",
+            "IsGurmukhi",
+            "IsHalfwidthandFullwidthForms",
+            "IsHangulCompatibilityJamo",
+            "IsHangulJamo",
+            "IsHangulSyllables",
+            "IsHanunoo",
+            "IsHebrew",
+            "IsHighPrivateUseSurrogates",
+            "IsHighSurrogates",
+            "IsHiragana",
+            "IsIPAExtensions",
+            "IsIdeographicDescriptionCharacters",
+            "IsKanbun",
+            "IsKangxiRadicals",
+            "IsKannada",
+            "IsKatakana",
+            "IsKatakanaPhoneticExtensions",
+            "IsKhmer",
+            "IsKhmerSymbols",
+            "IsLao",
+            "IsLatin-1Supplement",
+            "IsLatinExtended-A",
+            "IsLatinExtended-B",
+            "IsLatinExtendedAdditional",
+            "IsLetterlikeSymbols",
+            "IsLimbu",
+            "IsLowSurrogates",
+            "IsMalayalam",
+            "IsMathematicalOperators",
+            "IsMiscellaneousMathematicalSymbols-A",
+            "IsMiscellaneousMathematicalSymbols-B",
+            "IsMiscellaneousSymbols",
+            "IsMiscellaneousSymbolsandArrows",
+            "IsMiscellaneousTechnical",
+            "IsMongolian",
+            "IsMyanmar",
+            "IsNumberForms",
+            "IsOgham",
+            "IsOpticalCharacterRecognition",
+            "IsOriya",
+            "IsPhoneticExtensions",
+            "IsPrivateUse",
+            "IsPrivateUseArea",
+            "IsRunic",
+            "IsSinhala",
+            "IsSmallFormVariants",
+            "IsSpacingModifierLetters",
+            "IsSpecials",
+            "IsSuperscriptsandSubscripts",
+            "IsSupplementalArrows-A",
+            "IsSupplementalArrows-B",
+            "IsSupplementalMathematicalOperators",
+            "IsSyriac",
+            "IsTagalog",
+            "IsTagbanwa",
+            "IsTaiLe",
+            "IsTamil",
+            "IsTelugu",
+            "IsThaana",
+            "IsThai",
+            "IsTibetan",
+            "IsUnifiedCanadianAboriginalSyllabics",
+            "IsVariationSelectors",
+            "IsYiRadicals",
+            "IsYiSyllables",
+            "IsYijingHexagramSymbols",
+            "_xmlC",
+            "_xmlD",
+            "_xmlI",
+            "_xmlW",
+        };
+
+        public static bool IsEscapeCategory(string value)
+        {
+            return s_escapeCategories.Contains(value);
+        }
+
+        public static bool IsWordChar(char ch)
```

Comment:
this is the important function we needed.  if someone has high confidence on how this can be translated to usages of existing APIs in .net (i.e. methods on char or UnicodeCategory and whatnot), i can switch over to using that and can remove all this code.  @sharwell  as you sounded like you were very familiar with unicode.